app/helpers/link_helper.rb

```diff
@@ -1,13 +1,15 @@
 module LinkHelper
   EXTERNAL_LINK_CLASS = 'usa-link--external'.freeze
 
-  def new_window_link_to(name = nil, options = nil, html_options = {}, &block)
+  def new_window_link_to(name = nil, options = nil, html_options = nil, &block)
     html_options, options, name = options, name, capture(&block) if block
 
+    html_options ||= {}
     html_options[:target] = '_blank'
     html_options[:class] = [*html_options[:class], EXTERNAL_LINK_CLASS]
 
-    name = safe_join([name, content_tag('span', t('links.new_window'), class: 'usa-sr-only')], ' ')
+    name = ERB::Util.unwrapped_html_escape(name).rstrip.html_safe # rubocop:disable Rails/OutputSafety
```

Comment:
The logic here is essentially to try to recreate the same behavior of `safe_join`'s internal implementation, but allowing us to inject the `rstrip` between when the string is escaped and when it is marked as `html_safe`.

https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/actionview/lib/action_view/helpers/output_safety_helper.rb#L33-L37

It could also potentially help the performance slightly since we don't have to apply the same operation across both the `name` and added accessibility label, we're now only applying it to `name`, and assuming that the label would be marked as HTML-safe by default by `content_tag`.

Reply:
> It could also potentially help the performance slightly

I checked and the performance appears unchanged from #7455. It could also be offset by the addition of the `||=` assignment for `html_options`, as well as the new `rstrip` here.

Reply:
with the complexity (security/performance) of the safe_join/ERB calls, is it worth considering bringing back the <span></span><span></span>?

Reply:
I think I'd still prefer to avoid it, since...

- The previous behavior was still likely to cause some unpredictable behavior, specifically in how the block syntax would require the developer to know to apply their own `<span>` ([example](https://github.com/18F/identity-idp/pull/7455/files#diff-727c1af48666bc6f26041f22d5bca5b0f0ab95014c24d020434389879134191cL19))
- I'd like to keep the DOM tree as simple/shallow as possible, to reduce page size and browser processing. I don't think a plain wrapper element here or there is going to be too much an issue on its own, more a "death by thousand cuts" kind of thing.

Ultimately, I'd prefer if we could just have the spaces, and Safari would render them gracefully. Since this originates in USWDS, I'd want to check if it's improved in recent versions, or if there's some workaround to make it work better by default.

Reply:
Makes sense!